### DISCUSSION TITLE
How to create identity provider with openId/sml connect settings included over Rest Api   ?

### DISCUSSION THREAD

**Discussion by slnkykrn** [keycloak/keycloak#22539] [Q&A] [answered]
Hello , 
We want to seed identity providers with RestApi of Keycloak over c# code . But we can't pass OpendId connect settings or Saml Settings from code-base to the end point ? How should I do that ? I try to use ImportIdentityProvider endpoint, but for this one I can't see any example for the json part that I need to send ? 
Can you please help ? 

Thank you.

**Comment by justin-tay** [comment, marked as answer]
You could use the following REST API
* `POST` `/{realm}/partial-export`
* `POST` `/{realm}/partialImport`

You can determine the payload to send by just performing the operations in the browser with the developer console (F12) open and look at the network tab.

For instance here you could go to `Realm settings` then `Action` and select `Partial import` or `Partial export` to see what the payload looks like.

Alternatively you could use something like 
* `POST` `/{realm}/identity-provider/instances`

Here you can go to `Identity providers` and select `Add provider` to see the payload.

**Comment by slnkykrn** [comment]
Thank you so much , 
I tried with postman with below json over /{realm}/partialImport endpoint and I can successfully add identity providers with related Uris. 
`  "identityProviders": [
    {
      "alias": "saml",
      "displayName": "saml myid",
      "internalId": "dc874671-52fe-4568-ad01-2911646457fd",
      "providerId": "saml",
      "enabled": true,
      "updateProfileFirstLoginMode": "on",
      "trustEmail": false,
      "storeToken": false,
      "addReadTokenRoleOnCreate": false,
      "authenticateByDefault": false,
      "linkOnly": false,
      "firstBrokerLoginFlowAlias": "first broker login",
      "config": {
        "validateSignature": "false",
        "signingCertificate": "xxxx",
        "postBindingLogout": "true",
        "singleLogoutServiceUrl": "xxxxxxx",
        "postBindingResponse": "true",
        "nameIDPolicyFormat": "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress",
        "entityId": "http://localhost:8080/realms/master",
        "backchannelSupported": "false",
        "signSpMetadata": "false",
        "wantAssertionsEncrypted": "false",
        "idpEntityId": "urn:xxxxxx",
        "loginHint": "false",
        "allowCreate": "true",
        "wantAssertionsSigned": "false",
        "enabledFromMetadata": "true",
        "postBindingAuthnRequest": "true",
        "forceAuthn": "false",
        "wantAuthnRequestsSigned": "false",
        "attributeConsumingServiceIndex": "0",
        "allowedClockSkew": "0",
        "addExtensionsElementWithKeyInfo": "false",
        "principalType": "Subject NameID"
      }
    }
  ]`
